### PR TITLE
Warn if user is not using a supported platform

### LINKED ISSUE #5797: Check OS / pre-req's and fail fast if not met
# 🚀 Feature Request

### Description

A lot of time and hassle is spent troubleshooting issues that boil down to not running on supported operating systems, or not having the necessary pre-requisites (e.g. Homebrew on macOS).  It may be helpful to add in a check for this info by default, and then to introduce a new flag to bypass the check for people who are tinkering with unsupported configurations.

### Describe the solution you'd like

Example error messages:
```
ERROR: System does not meet the minimum Operating System requirement, use --bypass-os-check to skip this check. (expected Mac OS X 10.14+, found 10.13) 
ERROR: System does not meet software pre-requisites, use --bypass-prereq-check to skip this check. (Homebrew not found).
```
### Describe alternatives you've considered

Docs have all the info, but people don't RTFM.  Current error messages / stack dumps are quite scary.

### PR DESCRIPTION
**Pull request by dv8silencer**
**What type of PR is this?**

> Feature

**What does this PR do? Why is it needed?**
Early in the invocation of the clients, this PR will make it so that there is warn-level output if the user's platform is not supported.

**Which issues(s) does this PR fix?**

Fixes #5797 

**Other notes for review**

### CODE REVIEW

File: shared/prereq/prereq.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			}
+			// Otherwise we have a match between runtime and our list of accepted platforms
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// WarnIfNotSupported warns if the user's platform is not supported
+func WarnIfNotSupported(ctx context.Context) error {
```
Comment (farazdagi) [inline shared/prereq/prereq.go:98]: I think this method should not return anything, and consequently, if error occurs, it is logged, but upstream client (node's code) will continue as w/o interruption.

File: shared/prereq/prereq.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	supported, err := meetsMinPlatformReqs(ctx)
+	if err != nil {
+		return err
```
Comment (farazdagi) [inline shared/prereq/prereq.go:101]: So, instead of returning error, log a warning, and return.

File: beacon-chain/node/node.go [hunk 3]
```diff
@@ -91,6 +92,12 @@ func NewBeaconNode(cliCtx *cli.Context) (*BeaconNode, error) {
 		return nil, err
 	}
 
+	// Warn if user's platform is not supported
+	err := prereq.WarnIfNotSupported(cliCtx.Context)
+	if err != nil {
+		return nil, err
```
Comment (farazdagi) [inline beacon-chain/node/node.go:98]: We should not stop execution on error (please follow the comments in `WarnIfNotSupported` func). Warn if not supported, should just warn (if system version is not detectable, no error, warn that system version cannot be determined).

**Review by farazdagi — CHANGES_REQUESTED** [review]
Looks great, the only thing that seems to be off is stopping node execution if there is some issue in detecting the OS type/version. 

The scope of this PR should be limited to providing warning/guidance to user, not amending ability to run nodes.

File: shared/prereq/prereq.go (hunk 1 above)
Comment (dv8silencer) [reply shared/prereq/prereq.go:98]: Thanks.  Will address in upcoming commit

File: shared/prereq/prereq.go (hunk 2 above)
Comment (dv8silencer) [reply shared/prereq/prereq.go:101]: Thanks. Will address in upcoming commit

File: beacon-chain/node/node.go (hunk 3 above)
Comment (dv8silencer) [reply beacon-chain/node/node.go:98]: Thanks. Will address in upcoming commit

**Review by farazdagi — APPROVED** [review]
Amazing work, thank you!

**Pull request merged**